src/core/function_subscription_migrations.tf

```diff
@@ -0,0 +1,83 @@
+locals {
+  app_settings_commons = {
+    FUNCTIONS_WORKER_RUNTIME       = "node"
+    WEBSITE_NODE_DEFAULT_VERSION   = "14.16.0"
+    FUNCTIONS_WORKER_PROCESS_COUNT = 4
+    NODE_ENV                       = "production"
+
+    // Keepalive fields are all optionals
+    FETCH_KEEPALIVE_ENABLED             = "true"
+    FETCH_KEEPALIVE_SOCKET_ACTIVE_TTL   = "110000"
+    FETCH_KEEPALIVE_MAX_SOCKETS         = "40"
+    FETCH_KEEPALIVE_MAX_FREE_SOCKETS    = "10"
+    FETCH_KEEPALIVE_FREE_SOCKET_TIMEOUT = "30000"
+    FETCH_KEEPALIVE_TIMEOUT             = "60000"
+  }
+
+  // As we run this application under SelfCare IO logic subdomain, 
+  //  we share some resources
+  app_context = {
+    resource_group   = azurerm_resource_group.selfcare_be_rg
+    app_service_plan = azurerm_app_service_plan.selfcare_be_common
+    snet             = module.selfcare_be_common_snet
+  }
+
+}
+
+module "function_subscriptionmigrations" {
+  source = "git::https://github.com/pagopa/azurerm.git//function_app?ref=v2.1.0"
+
+  application_insights_instrumentation_key = data.azurerm_application_insights.application_insights.instrumentation_key
+  location                                 = var.location
+  name                                     = format("%s-subscription-migrations-fn", local.project)
+  resource_group_name                      = local.app_context.resource_group.name
+  subnet_id                                = local.app_context.snet.id
+  tags                                     = var.tags
+  allowed_ips                              = local.app_insights_ips_west_europe
+  allowed_subnets = [
+    data.azurerm_subnet.azdoa_snet[0].id,
+  ]
+
+  app_service_plan_id = local.app_context.app_service_plan.id
+  health_check_path   = "api/v1/info"
+  internal_storage = {
+    "blobs_retention_days" : 1,
+    "containers" : [],
+    "enable" : true,
+    "queues" : []
+  }
```

Comment:
```suggestion
  internal_storage = {
    "enable"                     = true,
    "private_endpoint_subnet_id" = data.azurerm_subnet.private_endpoints_subnet.id,
    "private_dns_zone_blob_ids"  = [data.azurerm_dns_zone.privatelink_blob_core_windows_net.id],
    "private_dns_zone_queue_ids" = [data.azurerm_dns_zone.privatelink_queue_core_windows_net.id],
    "private_dns_zone_table_ids" = [data.azurerm_dns_zone.privatelink_table_core_windows_net.id],
    "queues"                     = [],
    "containers"                 = [],
    "blobs_retention_days"       = 1,
  } 
```

Also u need to add data to query:
- private_endpoint_subnet_id
- private_dns_zone_blob_ids
- private_dns_zone_queue_ids
- private_dns_zone_table_ids

```suggestion
# data.tf get private_endpoint_subnet_id 
data "azurerm_subnet" "private_endpoints_subnet" {
  name                 = "pendpoints"
  virtual_network_name = format("%s-vnet-common", local.project)
  resource_group_name  = format("%s-rg-common", local.project)
}

# private_dns_zones.tf
data "azurerm_dns_zone" "privatelink_blob_core_windows_net" {
  name                = "privatelink.blob.core.windows.net"
  resource_group_name  = format("%s-rg-common", local.project)
}

data "azurerm_dns_zone" "privatelink_queue_core_windows.net" {
  name                = "privatelink.queue.core.windows.net"
  resource_group_name  = format("%s-rg-common", local.project)
}

data "azurerm_dns_zone" "privatelink_file_core_windows.net" {
  name                = "privatelink.file.core.windows.net"
  resource_group_name  = format("%s-rg-common", local.project)
}

data "azurerm_dns_zone" "privatelink_table_core_windows.net" {
  name                = "privatelink.table.core.windows.net"
  resource_group_name  = format("%s-rg-common", local.project)
}

```

Reply:
done in https://github.com/pagopa/io-infra/pull/77/commits/a40ccb10ddc544516c3289c02ecad4e942d399b7

Reply:
@balanza my bad is azurerm_private_dns_zone not azurerm_dns_zone.

also I see a typo `privatelink_table_core_windows.net` -> `privatelink_table_core_windows_net`